src/Plugin/media/Source/Media.php

```diff
@@ -519,4 +535,278 @@ private function getMediaValue(MediaInterface $media, $attribute_name) {
+      '#type' => 'select',
+      '#title' => $this->t('Media image bundle'),
+      '#default_value' => $this->getConfiguration()['media_image_bundle'],
+      '#options' => $this->getImageMediaBundles(),
+      '#empty_option' => $this->t('- Select -'),
+      '#description' => $this->t('To have the thumbnail mapped to a media entity rather than a file (the default) choose a media type and field to map the thumbnail to.'),
+      '#ajax' => [
+        'callback' => [$this, 'mediaImageBundleOnChange'],
+        'event' => 'change',
+        'wrapper' => 'media-image-field',
+      ],
+    ];
+
+    if (!empty($this->getConfiguration()['media_image_bundle'])) {
+      $form['media_image_field'] = $this->getMediaImageFieldElement($this->getConfiguration()['media_image_bundle']);
+    }
+    else {
+      $form['placeholder_media_image_field'] = [
+        '#markup' => '<div id="media-image-field"></div>',
+      ];
+    }
+
+    $form['media_image_entity_reference_field'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Media image entity reference field'),
+      '#default_value' => $this->getConfiguration()['media_image_entity_reference_field'],
+      '#options' => $this->getMediaImageEntityReferenceFieldOptions($entity),
+      '#empty_option' => $this->t('- Create -'),
+      '#description' => $this->t('Select the field to use to store the entity reference to the created media entity. If none is selected, one will be created for you.'),
+      '#states' => [
+        'invisible' => [
+          ':input[name="source_configuration[media_image_bundle]"]' => ['value' => ''],
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * AJAX callback for when the media bundle changes.
+   *
+   * Used to update the field selection.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array
+   *   The render array to render in place of the media image fields spot.
+   */
+  public function mediaImageBundleOnChange(array &$form, FormStateInterface $form_state) {
+    if ($media_image_bundle = $form_state->getValue(['source_configuration', 'media_image_bundle'])) {
+      $element = $this->getMediaImageFieldElement($media_image_bundle);
+    }
+    else {
+      $element = [
+        '#markup' => '<div id="media-image-field"></div>',
+      ];
+    }
+    return $element;
+  }
+
+  /**
+   * Get a list of media bundles that could be used to save an image.
+   *
+   * @return array
+   *   Array of media bundle labels keyed by bundle id, suitable for use in a
+   *   #type => select return array.
+   */
+  protected function getImageMediaBundles() {
+    // Get all media bundles.
+    $bundles = $this->entityTypeManager->getStorage('media_type')->loadMultiple();
+
+    $bundles = array_filter($bundles, function (MediaType $bundle) {
+      // Filter out any bundles that are use the same source plugin. That would
```

Comment:
Instead of "the same", how about "this source plugin"? I read this as multiple bundles using image, which would totally make sense.